### DISCUSSION TITLE
How to call a new parameter set

### DISCUSSION THREAD

**Discussion by SP171995** [pybamm-team/PyBaMM#2803] [Q&A] [answered]
I am using python version 3.8.3 and pybamm verison 22.1
I want to import new parameter set for Li-Ion cell, so i copied the Chen2020.py and renamed it like ABC.py and in simulation parameter_values = pybamm.ParameterValues('ABC')

i am giving this command but getting the error that FileNotFoundError: Could not find parameter,ABC.

my parameters_set.py file looks like this

import warnings
import pkg_resources
import textwrap
from collections.abc import Mapping

class ParameterSets(Mapping):
    """
    Dict-like interface for accessing registered pybamm parameter sets.
    Access via :py:data:`pybamm.parameter_sets`

    Examples
    --------
    Listing available parameter sets:

    .. doctest::

        >>> import pybamm
        >>> list(pybamm.parameter_sets)
        ['Ai2020', 'Chen2020', ...]

    Get the docstring for a parameter set:

    .. doctest::

        >>> import pybamm
        >>> print(pybamm.parameter_sets.get_docstring("Ai2020"))
        <BLANKLINE>
        Parameters for the Enertech cell (Ai2020), from the papers:
        ...

    See also: :ref:`adding-parameter-sets`

    """

    def __init__(self):
        # Load Parameter Sets registered to `pybamm_parameter_set`
        ps = dict()
        for entry_point in pkg_resources.iter_entry_points("pybamm_parameter_set"):
            ps[entry_point.name] = entry_point.load()

        self.__all_parameter_sets = ps

    def __new__(cls):
        """Ensure only one instance of ParameterSets exists"""
        if not hasattr(cls, "instance"):
            cls.instance = super(ParameterSets, cls).__new__(cls)
        return cls.instance

    def __getitem__(self, key) -> dict:
        return self.__all_parameter_sets[key]()

    def __iter__(self):
        return self.__all_parameter_sets.__iter__()

    def __len__(self) -> int:
        return len(self.__all_parameter_sets)

    def get_docstring(self, key):
        """Return the docstring for the ``key`` parameter set"""
        return textwrap.dedent(self.__all_parameter_sets[key].__doc__)

    def __getattribute__(self, name):
        try:
            return super().__getattribute__(name)
        except AttributeError as error:
            # For backwards compatibility, parameter sets that used to be defined in
            # this file now return the name as a string, which will load the same
            # parameter set as before when passed to `ParameterValues`
            if name in self.__all_parameter_sets:
                out = name
            else:
                raise error
            warnings.warn(
                f"Parameter sets should be called directly by their name ({name}), "
                f"instead of via pybamm.parameter_sets (pybamm.parameter_sets.{name}).",
                DeprecationWarning,
            )
            return out

#: Singleton Instance of :class:ParameterSets """
parameter_sets = ParameterSets()

**Comment by valentinsulzer** [comment, marked as answer]
It's easier to just pass the dictionary of parameter values in this case

```python
from abc import get_parameter_values

pybamm.ParameterValues(get_parameter_values())
```
where `get_parameter_values()` returns the dict of parameter values

**Reply by SP171995** [reply]
thanks, it solved the problem but now I have to parameter_vlaues.py by converting "value" to string, initially, it was float(value), but not sure now if it will have any side effects

**Comment by naltang1** [comment]
I encountered the same problem as you. 
I got the following message after running the above commend

-----------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-26-22bf928daff8> in <module>
----> 1 from ABC import get_parameter_values
ModuleNotFoundError: No module named 'ABC'

Could you help me how to solve this problem? 
I am using python version 3.8.8 and pybamm verison 23.2.1

**Reply by agriyakhetarpal** [reply]
Hi @naltang1, this looks like a naming and/or module location issue. Could you check if you have a file with your desired parameters called `ABC.py` in your directory?

**Reply by kratman** [reply]
To add the file to `pybamm/input/parameters/lithium_ion`, then you need to also update `pybamm/CITATIONS.bib` so that the cite command works, and update the list of parameter sets in `setup.py`

Once those three steps are done you should be able to use your parameter set just like the existing ones.
